### PR TITLE
kubelet: handle containers in the "created" state

### PR DESCRIPTION
**Pull request by yujuhong**

### CODE REVIEW

**Comment by yujuhong** [comment]
Kubelet doesn't really handle the "created" container state, and has been treated that as unknown. I think some changes in CRI implementation (kuberuntime) added the "created" state, causing kubelet to fail in some corner cases. One thing I observed is that if a docker container stays in the "created" state for whatever reason AND the restart policy is not always, kubelet will *not* restart the container at all. This PR should fix the problem.

**Comment by feiskyer** [comment]
LGTM

File: pkg/kubelet/kubelet_pods.go [hunk 1]
```diff
@@ -1210,6 +1210,13 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 		switch cs.State {
 		case kubecontainer.ContainerStateRunning:
 			status.State.Running = &v1.ContainerStateRunning{StartedAt: metav1.NewTime(cs.StartedAt)}
+		case kubecontainer.ContainerStateCreated:
+			// Treat containers in the "created" state as if they are exited.
+			// The pod workers are supposed start all containers it creates in
+			// one sync (syncPod) iteration. There should  not be any normal
```
Comment (Random-Liu) [inline pkg/kubelet/kubelet_pods.go]: Double space.

**Review by Random-Liu — COMMENTED** [review]
If so, the bug is introduced for half a year already... Haha

File: pkg/kubelet/kubelet_pods.go (hunk 1 above)
Comment (yujuhong) [inline pkg/kubelet/kubelet_pods.go]: fixed.

**Comment by yujuhong** [comment]
> If so, the bug is introduced for half a year already... Haha

It rarely happens. I think it happens if kubelet creates the containers, but get killed before starting the containers. After kubelet restarts, it'd think all containers have run and should not be restarted anymore. I hit this with my other PR.

**Comment by yujuhong** [comment]
Applying the lgtm label based on the comments above.

**Comment by yujuhong** [comment]
@k8s-bot cvm gce e2e test this

**Comment by yujuhong** [comment]
@k8s-bot cvm gce e2e test this

**Comment by yujuhong** [comment]
The previous cvm e2e run failed with `ERROR: (gcloud.components.install) The component [alpha] failed to download.`

**Pull request merged**